### PR TITLE
ISSUE-58: Correctly handle a part after parentheses

### PR DESCRIPTION
**Pull request by mrmlnc**
This is a partial fix of #58. Source for this case is https://github.com/mrmlnc/fast-glob/issues/263.

Changes:

* Add some tests for the `scan` method; I tried to collect all the edge cases.
* Correctly handle a case with a part after parentheses. E.g. `base/(a|b)/*.js`

I'm not sure how to test this correctly. Changes based on `extglob` support:

https://github.com/micromatch/picomatch/blob/66e1b200f3ea1784bdeae0770c8e5cd95f55e334/lib/scan.js#L169-L199.

### CODE REVIEW

**Comment by jonschlinkert** [comment]
Thank you! will review tonight

File: test/api.scan.js [hunk 1]
```diff
@@ -252,6 +262,56 @@ describe('picomatch', () => {
...
+      // assertParts('(a\\b)', ['(a\\b)']);
+      // assertParts('foo\\[a\\/]', ['foo\\[a\\/]']);
+      // assertParts('foo[/]bar', ['foo[/]bar']);
+      // assertParts('/dev\\/@(tcp|udp)\\/*\\/*', ['', '/dev\\/@(tcp|udp)\\/*\\/*']);
+
+      // Right now it returns ['*']
+      // assertParts('*/', ['*', '']);
+
+      // Right now it returns ['!(!(bar)', 'baz)']
+      // assertParts('!(!(bar)/baz)', ['!(!(bar)/baz)']);
```
Comment (jonschlinkert) [inline test/api.scan.js]: @mrmlnc are the commented out tests still passing?

Comment (mrmlnc) [reply test/api.scan.js]: These tests (here and above) have an incorrect result for the current implementation of the `scan` method.

I can delete them or move them to another `it` with `skip` or something.

**Comment by mrmlnc** [comment]
@jonschlinkert we can merge and ship it?

**Comment by mrmlnc** [comment]
@jonschlinkert friendly ping :)

**Comment by jonschlinkert** [comment]
Sorry for the delay! I've been a lot busier than usual for the past couple of weeks. I reviewed the code and everything looks good to me. 

I made sure that you're able to publish on NPM since you're already an owner on the org. Please feel free to publish! 

Thanks for the awesome PR!

**Comment by mrmlnc** [comment]
I'm sorry, I didn't know I had the rights to release versions.
